File: the_money_example_c++/test/money-test.cpp
#include "gtest/gtest.h"
#include "main/money.h"

TEST(MoneyTest, TestMultipulication) {
  Money five = dollar(5);
  ASSERT_EQ(dollar(10), five.times(2));
  ASSERT_EQ(dollar(15), five.times(3));
}

TEST(MoneyTest, TestEquality) {
  ASSERT_TRUE(dollar(5) == dollar(5));
  ASSERT_FALSE(dollar(5) == dollar(6));
  ASSERT_FALSE(franc(5) == dollar(5));
}

TEST(MoneyTest, TestCurrency) {
  ASSERT_EQ("USD", dollar(1).currency());
  ASSERT_EQ("CHF", franc(1).currency());
}

TEST(MoneyTest, TestSimpleAddition) {
  Money sum = dollar(5) + dollar(5);
  ASSERT_EQ(dollar(10), sum);
}

File: the_money_example/csharp/TheMoneyExample.Tests/MoneyTest.cs
using NUnit.Framework;

namespace Money;

public class MoneyTest
{
    [Test]
    public void TestMultiplication()
    {
        Money five = Money.Dollar(5);
        Assert.AreEqual(Money.Dollar(10), five.Times(2));
        Assert.AreEqual(Money.Dollar(15), five.Times(3));
    }

    [Test]
    public void TestEquality()
    {
        Assert.True(Money.Dollar(5).Equals(Money.Dollar(5)));
        Assert.False(Money.Dollar(5).Equals(Money.Dollar(6)));
        Assert.False(Money.Franc(5).Equals(Money.Dollar(5)));
    }

    [Test]
    public void TestCurrency()
    {
        Assert.AreEqual("USD", Money.Dollar(1).Currency());
        Assert.AreEqual("CHF", Money.Franc(1).Currency());
    }

    [Test]
    public void TestSimpleAddition()
    {
        var five = Money.Dollar(5);
        IExpression sum = five.Plus(five);
        var bank = new Bank();
        Money reduced = bank.Reduce(sum, "USD");
        Assert.AreEqual(Money.Dollar(10), reduced);
    }
}
